src/micro_helmet/provider.es

```diff
@@ -0,0 +1,48 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { Provider } from './context';
+import { parseProps, collectProps } from './utils';
+
+
+class MicroHelmetProvider extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.propsArray = [];
+  }
+
+  generateContext() {
+    const { context } = this.props;
+    context.value = parseProps(this.propsArray.reduce(collectProps, {}));
+    if (process.browser) this.onChangeInBrowser();
+  }
+
+  addProps(props) {
+    this.propsArray.push(props);
+    this.generateContext();
+
+    return () => {
+      this.propsArray = this.propsArray.filter((item) => item !== props);
+      this.generateContext();
+    };
+  }
+
+  onChangeInBrowser() {
+    const { title } = this.props.context.value;
+    if (title && title !== document.title) document.title = title;
+  }
+
+  render() {
+    return <Provider value={this}>{this.props.children}</Provider>;
```

Comment:
only expose the functions you use. create a staticContext object in constructor and pass it in here.